AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -814,6 +814,40 @@ - (void)setUsesImplicitHierarchyManagement:(BOOL)value
   _usesImplicitHierarchyManagement = value;
 }
 
+- (BOOL)_hasTransitionInProgress
+{
+    ASDN::MutexLocker l(_propertyLock);
+    return _transitionInProgress;
+}
+
+/// Starts a new transition and returns the transition id
+- (int32_t)_startNewTransition
+{
+  ASDN::MutexLocker l(_propertyLock);
+  _transitionInProgress = YES;
+    
+  if (!_transitionSentinel) {
+    _transitionSentinel = [[ASSentinel alloc] init];
+  }
```

Comment:
I think ASSentinel provides a somewhat useful but very thin layer of abstraction. It's tempting to remove it and use an atomic int instead.